commands/build.go

```diff
@@ -614,10 +620,15 @@ func decodeExporterResponse(exporterResponse map[string]string) map[string]inter
 	return out
 }
 
-func wrapBuildError(err error, bake bool) error {
+func wrapBuildError(err error, term bool, bake bool) error {
 	if err == nil {
 		return nil
 	}
+	if ebr, ok := err.(*build.ErrorWithBuildRef); ok {
+		return errors.Errorf("%v\n\n%s", ebr.Err, buildDetailsOutput(map[string]string{
```

Comment:
wrap function should not flatten the typed error into a string. This should happen when printing the error, currently handled in main, eg. same as the `errdefs.Sources()` handling.